src/inc-init.php

```diff
@@ -476,6 +476,10 @@
 if ($_CONF = $_DB->query('SELECT * FROM ' . TABLE_CONFIG, false, false)) {
     // Must be two-step, since $_CONF can be false and therefore does not have ->fetchAssoc().
     $_CONF = $_CONF->fetchAssoc();
+    if ($_CONF) {
+        // See if the database is read-only at the moment, due to an announcement with the configuration.
+        $_CONF['lovd_read_only'] = (bool) $_DB->query('SELECT COUNT(*) FROM ' . TABLE_ANNOUNCEMENTS . ' WHERE start_date <= NOW() AND end_date >= NOW() AND lovd_read_only = 1')->fetchColumn();
```

Comment:
It seems a bit strange to check for the read-only status this way, compared to having the field `lovd_read_only` part of the config table.

On the other hand, this way the read-only status is terminated automatically when the announcement ends. So I guess we could leave it like this.